app/src/main/java/io/github/hidroh/materialistic/WebFragment.java

```diff
@@ -536,6 +573,19 @@ void onItemLoaded(@NonNull Item response) {
         bindContent();
     }
 
+    private void downloadFileAndRenderPdf() {
+        mDownloadModule.downloadFile(getContext(), mItem.getUrl(), PDF_MIME_TYPE, new DownloadModule.DownloadModuleCallback() {
+            @Override
+            public void onFailure(IOException e) {
+            }
+
+            @Override
+            public void onSuccess(File file) throws IOException {
+                mWebView.post(() -> reloadUrl(PDF_LOADER_URL, file));
```

Comment:
If we change the downloader to fire callback on main thread like in my other comment then we don't need to wrap this with `post()`.

Reply:
Yup